staging/src/k8s.io/client-go/tools/clientcmd/config.go

```diff
@@ -75,8 +77,16 @@ func (o *PathOptions) GetLoadingPrecedence() []string {
 	if envVarFiles := o.GetEnvVarFiles(); len(envVarFiles) > 0 {
 		return envVarFiles
 	}
-
-	return []string{o.GlobalFile}
+	files, err := loadFileOrDir(o.GlobalFile)
```

Comment:
This looks like it should be the same logic as `getRecommendedHomeFileOrDir` with the file and dirname passed through.